### DISCUSSION TITLE
How to visualize throats in the Paraview

### DISCUSSION THREAD

**Discussion by GaoC-en** [PMEAL/OpenPNM#2367] [Q&A]
Dear professor:
I have learned the tutorials and I wanted to visualize the results in Paraview.  I read the article at this website, https://github.com/PMEAL/OpenPNM/blob/dev/examples/getting_started/view_vtp_in_paraview.md, but I can not deal with it. 
In my paraview, I can not find the Tube Filter in the Shrink Filter such like you showed in the website above. Besides that, I used Glyph to see the throats, but there is only one direction. How can I deal with this problem? 

Sincerely.

**Comment by jgostick** [comment]
Showing throats as tubes has always been a HUGE pain.  We've investigated it pretty deeply and there is no easy way.  Mostly we just don't bother showing the throats as tubes.  Anyway if that tutorial above didn't work then I have no other suggestions.  BTW....which version of paraview do you have?  I doubt they removed the shrink filter, it's under the filters/alphabetical menu.

**Reply by GaoC-en** [reply]
Thanks for your instant replying. I used the version 5.9.0, I can find the Shrink filter but when I wanted to use tube filter in Shrink, it can not be found. Could you please tell me the version you used in the tutorial? Besides that, I find that the vtp file that I exported don't contain the orientation information, could it be the reason of the same direction of all throats?

**Reply by jgostick** [reply]
Hi @GaoC-en 
I followed the instructions on that page using Paraview 5.9.1 and got this:

![demo](https://user-images.githubusercontent.com/5092049/162246261-46cb5724-6f17-450a-bc03-12cf5eed6f65.png)
:

**Reply by GaoC-en** [reply]
Thank you sir, it is so kind of you to make it again. Could you please show me the website to download the (run_script.py) file? I can not find it on the Github. My screenshot is shown as followed, there is no Tube in my filters. I would like to know whether it is caused by the absence of orientation information?
![dd11d95d558ee325c10933bf55b2b1f](https://user-images.githubusercontent.com/75919331/162348167-974a1075-67a5-41dc-94a1-d6911da2e2c5.jpg)

**Reply by jgostick** [reply]
It's not caused by the orientation information, or lack of.  I don't know why the Tube filter is not showing, but we are using exactly the same version of paraview (5.9.1). Have you clicked "apply" to all the other filters as you applied them?  Without this it's impossible to add the next filter.  In my install the unavailable filters are grey, but perhaps yours are set to be not shown if they are not available?  I really can't think of anything else it could be.  You could also try to uninstall and reinstall paraview, maybe try another version?

**Reply by GaoC-en** [reply]
Dear professor, I think I have found the reason. Paraview showed that the input data must be vtkPloyData, I exported the model through the code below:
`op.io.VTK.export_data(network=pn,` phases=water, filename='46')`
And my exported file is '**.vtp; file. Is there any mistake I have done? Thanks a lot for your help!
![64b9e066fadd8d561b087405ec21321](https://user-images.githubusercontent.com/75919331/162618385-2110aff8-1af1-456b-87af-bab86b074ea0.jpg)

**Reply by jgostick** [reply]
VTK stands for [visualization tool kit](https://vtk.org/) which is made by the [Kitware group](https://www.kitware.com/).  Paraview is built on VTK. "VTK" is the generic name of all files that work with this toolkit, but there are some specific file types like VTU and VTP.  To visuzalize openpnm networks the data needs to be in a VTP file.  When you call the ``VTK class`` in OpenPNM it outputs a VTP file.  

I have just just tested this again, but ensured that I was using OpenPNM version 2.8.2, which is the latest release on conda and probably the one you're using, and I can confirm again that it JUST WORKS.  I don't know why you're having trouble, but I really can't afford to spend any more time giving you tech support for this.  

If you can't figure this out, I think you should just give up on visualizating the throats.  They only get in way...we almost never bother to include the throat tubes.
